src/mocks/data/draft_reports_object.json

```diff
@@ -0,0 +1,50 @@
+{
+  "status": "processing", // enum: one of ["processing", "failed", "delivered"]
+  "current_stage": "dex-upload",
+  "current_action": "metadata-validation",
+  "file_name": "test.txt",
+  "transport_id": "unique_guid", //represents the id for the entire transaction i.e. upload_id
+  "timestamp": "12-12-2023", -- first initial timestamp
+  "data_stream_id": "aims-celr",
+  "data_stream_route": "csv",
+  "reports": [
+    {
+    "stage": "DEX HL7 v2", //describes a chunk of the pipeline, usually associated with a specific product; string could be human readable, or transformed from id to human readable
+    "action": "debatch", //describes specific thing that happened as part of the stage
+    "schema_name": "dex-hl7",
+    "schema_version": "1.0",
+    "data_stream_id": "aims-celr",
+    "data_stream_route": "csv",
+    "jurisdiction": "MD",
+    "transport_id": "unique_guid", //represents the id for the entire transaction i.e. upload_id
+    "parent_id": "unique_guid", (would be null if it is the parent?) //id for the thing that resulted in this report?
+    "message_id": "unique_guid",//id for the thing this report is about
+    "type": "failure", //represents machine readable way to understand what happened in the stage action; enum: one of ["failure", "success", "pending"] ("delivered?")
+    "timestamp_complete": "12-12-2024", //time action completed
+    "message": "Debatch failed", //Action summary. For success would read "Debatch successful"
+    "errors": [
+      {
+      "type": "validation",
+      "stage": "dex-upload",
+      "action": "file-upload",
+      "level": "warning",
+      "message": "There was a problem. Here's an actionable message",
+      "details": {"key": "value"}
+      }, {...}
+    ],
+    "routing_metadata?": { all the specific submission metadata }
+    "data": { unique data to the report? },
+    }, {...}
+  ],
+  "errors": [ //rolled-up errors for observability. might require specialized logic for meaningful debatched message errors/warnings
```

Comment:
If we are able to extract errors to put at the root level, we should also associate these errors with a report ID to ensure they can be tied to a specific report. I'm thinking about a scenario where we have several hl7 warnings across one or many message which do not prevent the file from progressing through the pipeline. In these cases, reports on downstream stages will continue to be produced while identified errors will likely persist. Here is an example of what this might look like (Note how the action in the error is from a previous step than the current processing action, and the error is at a level of warning):

```
[
  {
    "status": "processing",
    "current_stage": "DEX HL7",
    "current_action": "Transform JSON Lake",
    "file_name": "filename-with-fixed-chars.hl7",
    "transport_id": "d15a368e-56a7-4c4d-b109-eab15e59a2b7",
    "timestamp": "12-12-2023",
    "data_stream_id": "aims-celr",
    "data_stream_route": "hl7",
    "reports": [...],
    "errors": [
      {
        "type": "validation",
        "stage": "DEX HL7",
        "action": "Validate HL7 Structure",
        "level": "warning",
        "message": "Segment OBX (OBX - observation/result segment) has extra children",
        "details": {
          "line": 6,
          "column": 1,
          "path": "OBX[2]",
        }
      },{...}
    ]
  } 
]
      ```

Reply:
The rolled up errors should be a summary. Files and messages should be separate yet related resources.
Consider how we separate message errors from file errors.
Need an id to connect the error message to the report.
All rolled up errors in a file report, should be at the file level.
All rolled up errors for messages , should be at the message level